src/lib/plot/axis/x-axis.js

```diff
@@ -31,14 +32,21 @@ const propTypes = {
   ])
 };
 
+delete propTypes.axisType;
```

Comment:
I'm not sure I understand what you are suggesting with rest parameters? I also haven't heard of delete preventing optimizations? (Though I would be excited to see source saying otherwise)